### PR TITLE
Chargepoint API Driver Version 1.0

### LINKED ISSUE #994: ChargePoint API Driver
Kisensum is developing a driver interface to the ChargePoint Web Services API.  It will support the retrieval of basic charging telemetry and the management of charging power limits.  It will be based on the 5.0 version of the ChargePoint API.  The 4.1 version is described [here](https://na.chargepoint.com/UI/downloads/en/ChargePoint_Web_Services_API_Guide_Ver4.1_Rev4.pdf).

We will publish a design document by November 9 and plan to make a first release available by the end of November.

### PR DESCRIPTION
**Pull request by ffreckle**
The Chargepoint API Driver for VOLTTRON enables real-time monitoring
and control of Chargepoint EVSEs within the VOLTTRON platform
by creating a standard VOLTTRON device driver on top of the
Chargepoint Web Services API.  Each managed chargestation
looks like a standard VOLTTRON device, monitored and controlled
through the VOLTTRON device driver interface. Resolves #994.

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1074)

### CODE REVIEW

**Comment by ffreckle** [comment]
@jhaack et al:  We have a working Chargepoint API driver and are ready to review.  Please let me know when you have time to discuss.  Thanks.

**Comment by jhaack** [comment]
@kmonson please take a look

File: examples/configurations/drivers/chargepoint1.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "driver_config": {
+        "stationID" : "1:34003",
+        "username" : "9d44ba0be5fe6b6628e50af1335d4fcf5743a6f3c63ee1464051443",
+        "password" : "**Put valid Chargepoint password here.**"
+    },
+    "campus": "campus",
+    "building": "building",
+    "unit": "station_001_01",
```
Comment (kmonson) [inline examples/configurations/drivers/chargepoint1.config]: The "campus", "building", and "unit" values are now ignored by the driver framework. We now use the name of the configuration file when it's put in the config store to derive the device topic .

File: examples/configurations/drivers/chargepoint2.config [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "driver_config": {
+        "stationID" : "1:34033",
+        "username" : "9d44ba0be5fe6b6628e50af1335d4fcf5743a6f3c63ee1464051443",
+        "password" : "**Put valid Chargepoint password here.**"
+    },
+    "campus": "campus",
+    "building": "building",
+    "unit": "station_001_02",
```
Comment (kmonson) [inline examples/configurations/drivers/chargepoint2.config]: The "campus", "building", and "unit" values are now ignored by the driver framework. We now use the name of the configuration file when it's put in the config store to derive the device topic. These should be removed.

**Comment by ffreckle** [comment]
@kmonson We've made the changes that were discussed in our call and added a section for Chargepoint Driver configuration.  We  should be good to merge.  Thanks Kyle.

**Comment by kmonson** [comment]
OK, I'll take a look at it tomorrow.

**Comment by kmonson** [comment]
Looks good!

**Pull request merged**